sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -4,54 +4,244 @@
 #nullable enable
 
 using System;
+using System.Collections;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq.Expressions;
 using System.Reflection;
 
 namespace Azure.Core.Pipeline
 {
+#pragma warning disable CA1001 // Implement IDisposable
     internal readonly struct DiagnosticScope : IDisposable
+#pragma warning restore CA1001 // Implement IDisposable
     {
-        private readonly DiagnosticActivity? _activity;
+        private static readonly ConcurrentDictionary<string, object?> ActivitySources = new();
```

Comment:
How do you prevent this from growing out of control? I don't see any obvious eviction.

Reply:
We add one entry per client type.